### ISSUE TITLE
Flow with exact types and subresolvers produces unusable types

### ISSUE THREAD

**Issue by robvk** [dotansimha/graphql-code-generator#1673]
**Describe the bug**
When using the exact types option the intersection way of writing the flow types creates unusable types as any object will fail. This is a flow issue with how exact objects interact with the '&' operator.

It is shown best when using a fragment:

**To Reproduce**

1. My GraphQL schema:

```graphql
schema {
  query: Query
}

type Query {
  user(id: ID!): User
}

type Address {
  street: String!,
  city: String!
}

type User {
  id: ID!
  email: String!
  address: Address!
}
```

2. My GraphQL operations:

```graphql
fragment UserFields on User {
  id
  email
  address {
    street
    city
  }
}
```

3. My `codegen.yml` config file:

```yml
schema: ./schema.json
documents: ./documents/*.js
generates:
  ./types.flow.js:
    plugins:
      - flow
      - flow-operations
    config:
      skipTypename: true
      useFlowExactObjects: true
```

**Expected behavior**
This generates the following typing for the fragment:

```
export type UserFieldsFragment = ($Pick<User, {| id: *, email: * |}> & {| address: $Pick<Address, {| street: *, city: * |}> |});
```

Unfortunately this type is unusable because of a [bug in flow](https://github.com/facebook/flow/issues/6680) with intersection and exact typing. If I for example want to make an object with that type I will always get an error.

This [flow try](https://flow.org/try/#0C4TwDgpgBAJACgSwMYGsA8B5ATgg5ggOwC4oMAjAKwiWABooBpCEAZxPKpoD4oBeWDgFkAhmARomrehOZcAFChKSAlHx4wAogBsIAWwgFgAFXARMOfAXqSuXANwAoKM6gPQkKAFUWELADEECC0AExY-LGFcfUM+KDl4ZHRvX3oAbwAfKARgkgAqej1hBC08qHSAXx4AMigMqGFg4KwIFjZYRFQ0AEFG5ta0zJZgZohgPPokBFBSip4K5UcXVwB6Zah3aGT-QJCwiKiDYFi5VKgAOguEzq2BrJyofKhC4pnK+gamlrbTi7Or9B6n36tUGwwgo3GUEm0weZUq5XKCycLgcblMUEBfRYsVSyKGIzGUHxhFwtGR0JAJGJBFJDnKyOcaI8WxxyOyVOGJLJzmeJSJnJp9GRHyxJExXzpqKQAHsCEN1i1gBwSFsAkFQuFItEjvxcc52VAAOQARkN3KeuiKfMNLGl+llEAAAlErWcZbozcLel8SHqXPjwYTDQBlO3QAOjT1LCkkENhqFTEBRqD0+lAA) hopefully explains the issue. On line 5 I have added an alternative way of defining it in a comment that could work as a solution.

**Comment by dotansimha** [comment]
@robvk I managed to reproduce it, but I'm not sure that your solution is the right one. 
Using your solution, it seems to work, but it's not really an exact type, [try this one](https://flow.org/try/#0C4TwDgpgBAqgzhATgMQJYQDYBM7MQQwHMBbCAO2CgF4oAKAbygDoWASABVQGMBrAHnhIANFHoAfKKiwAuKACoREYvlQZZcqGIC+APhH4sWRBDhxZjFkw7d+AQUPHTI8VDjBjEYOpFdUodZq6WloAlADcAFBQ0REA9BoAAgBmGAD2AO7ysRFR0VDxGrYYGFAARgCuqsAAtKhkUPhkWFBc5W6pxK5c+Bj4iHAiymCQzcCpUMAAFhCoiA1cwOU9UABuPeUmWbnREAAeYKmIlKCQUADK3b391KLbeQCSACKyboh1hEJ30Wfu7y+-ZA+XygACFUqkMBBGrJSuDIY1PnkHhRZGRysRSsJgcg0vgvFA0RisUitJEkXc9gcjhNwNB7EYTHAbvRga8IJ5ZKwAKKQ0gUAAqtL4Fx6fQGUAA5D83oCJXpgb5-FBubzyMBBZBhZcxSIpQDCHLEXlSXcKftDsdaVAAIobRAgZnAtpIAD8shdgkQRuiJvJfrylItNNOtqQIE9tkQhCZNBZSOiUk5PKUao1EC1ov6uqecruvryZqpltOnsd8ckMmVyb56qFIqu4olOfl5aUKjUVdVArr2qzkul70NwIMDNMsnpjgGebJBaRJ2g1l4fAA8m9CHVZMvSgArCALEQAaQgIDMUC3u4WOhurHPAFl8GBUHwjyeRM-jzpaDxZC+QtQryqKbdpqq6oOuZCHh+OiRHcXCpGQbgTCYwDnrInpoJgOB4EQNZluW8aJpKACMErevhOzKKosgSnAHQQPBEAJCQ7ZMHBxCkcC5YjpO5icfhbIcv2dGuO47LABx5HloqIDUWcwnSRJklzshsgAMxkeRWgaeRsKpGAsgAEx8T6QA).

**Comment by robvk** [comment]
You are absolutely right. 

Should be like [this](https://flow.org/try/#0C4TwDgpgBAqgzhATgMQJYQDYBM7MQQwHMBbCAO2CgF4oAKAbwB8oA6NgEgAVUBjAawA88JABooTKKiwAuKACoxEYvlQZZcqIwC+APjH4sWRBDhxZ9Vh278BAQUPHTYiXGDGIwdWJ6pQ6zbpa2gCUANwAUFBR4QD0GgACAGYYAPYA7vIx4ZFRUHEathgYUABGAK6qwAC0qGRQ+GRYUDxlrinEUHA8+Bj4iHBiymCQTcApUMAAFhCoiPU8wGU9UABuPWUmmTlREAAeYCmIlKCQUADK3b391OLbuQCSACKyroi1hCJ3UWdu7y+-ZA+XygACEUikMBAGrISuDIQ1PrkHhRZGQysQSqJgchUvhPFA0RisUitBEkXc9gcjhNwNB7EYTHAbvRga8IB5ZOwAKKQ0gUAAqtIEFx6fQGUAA5D83oCJXpgT4-FBubzyMBBZBhZcxWIpQDCHLEblSXcKftDsdaVAAIobRAgZnA1pIAD8shdwkQRqiJvJftylItNNOtqQIE9tkQhCZNBZSKiUk5PKUao1EC1ov6uqecruvtyZqpltOnsd8ckMmVyb56qFIqu4olOfl5aUKjUVdVArr2qzkul70NwIMDNMsnpjgGebJBaRJ2gXF4ggA8m9CLVZMuSgArCALMQAaQgIDMUC3u4WOhu7HPAFl8GBUAIjyexM-jzpaHxZC-gtQryqKbdpqq6oOuZCHh+OgRHcPApGQrgTCYwDnrInpoJgOB4EQNZluW8aJpKACMErevhOzKKosgSnA7QQPBEDxCQ7YsHBxCkcC5YjpO5icfhbIcv2dGdG47LABx5HloqIDUWcwnSRJklzshsgAMxkeRWgaeRsIpGAsgAEx8T6QA) then.

**Comment by dotansimha** [comment]
Spread implemented in: https://github.com/dotansimha/graphql-code-generator/pull/2602

**Comment by dotansimha** [comment]
Fixed in 1.8.0 🚀
